src/client/components/scatter_plot/view.tsx

```diff
@@ -1,5 +1,256 @@
+import { observer } from 'mobx-react'
+import * as Plotly from 'plotly.js'
 import * as React from 'react'
+import { inject } from '../../../inversify.config'
+import { MenuBar } from '../menu_bar/view'
+import { ScatterplotController } from './controller'
+import * as style from './style.css'
 
-export const Scatter = () => (
-  <h1>Scatter</h1>
-)
+interface Trace {
+  mode: string
+  type: string
+  hoverinfo: string
+  marker: Plotly.ShortMarker
+  name: string
+  xVal: number
+  id: number
+  display: boolean
+  addTrace: boolean
+}
+
+interface DataPoint {
+  label: string
+  values: number[]
+}
+
+@observer
+export class ScatterplotView extends React.Component<any, any> {
```

Comment:
This component needs some refactoring to be inline with some guidelines which I haven't done a good job of documenting yet 🙂 

Some goals I had when writing the localisation component:

  - If possible, they should just render, only using the props they were given.
  - Preferably they have very little "logic".
  - Generally this means they become almost "stateless", i.e. there are a minimal number of member fields in the view itself.
  - Controllers were just responsible for responding to user actions, and appropriately manipulating the state. They know nothing about the DOM, e.g. they should not be given an HTML element for instance, or directly call `window.someGlobalThing`.
  - If you take [`LocalisationView`](https://github.com/NUbots/NUsight2/blob/master/src/client/components/localisation/view.tsx) as an example, it is only concerned about how things look, and propagating view events (`onClick`, etc) out to the controller. Views never directly manipulates data.

These ideas come down to some rules of thumb which roughly align with standard [MVC](https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93controller):
- Views should as stateless.
- Models should contain all your state.
- Controllers should do all your logic.

To achieve this for the scatterplot component, I think we would need to:
- Moving all your state, e.g. `traces`, `nextTraceId`, `graphUpdateX`, `graphUpdateY`, `graphUpdateZ`, `maxPoints`, `fps`, into a separate `ScatterPlotModel`.
- Moving any logic into the controller
- Remove any view/dom related things from the controller, instead manipulate the state which should trigger a view change.